exercises/concept/lasagna/src/test/java/utils/ReflectionProxy.java

```diff
@@ -0,0 +1,477 @@
+package utils;
+
+import java.lang.reflect.*;
+
+import static java.lang.Class.forName;
+
+public abstract class ReflectionProxy {
+
+    /**
+     * An instance of the target class (if found)
+     */
+    private Object target;
+
+    /**
+     * A constructor to instantiate the target class with parameters
+     * @param args An array of parameters matching the constructor from the target class
+     */
+    public ReflectionProxy(Object... args) {
+        this.target = instantiateTarget(args);
+    }
+
+    /**
+     * The default constructor, for when you have already an instance of the target class
+     * @param target An instance of the target class
+     */
+    public ReflectionProxy(Object target) {
+        this.target = target;
+    }
+
```

Comment:
It doesn't technically matter for instantiation but would it be more valuable to change the constructors to `protected` to control scope?

Reply:
Were there two constructors?

Reply:
The constructor that receives the target instance is not being used. Moved to the "unused" region 👍

Reply:
Ok, the main constructor is still marked `public`.

Reply:
I think that's ok, since the class is abstract, right?

Reply:
> It doesn't technically matter for instantiation but would it be more valuable to change the constructors to protected to control scope?

It's more of a style thing since this may be used as a template in further exercises.

Reply:
ok, done!